libraries/AP_Camera/AP_Camera.cpp

```diff
@@ -193,6 +209,8 @@ void AP_Camera::init()
         return;
     }
 
+    memset(_backends, 0, sizeof(void*)*AP_CAMERA_MAX_INSTANCES);
```

Comment:
Why is this being done?!
```suggestion
```

Reply:
Because the array is not zeroed and we check whether the backend has been allocated.

Reply:
that is a surprise, as anything not stack allocated should start as zero. It would be worth trying to find out why it is non-zero, we rely on that pretty heavily

Reply:
Ok I removed it and it still seems to work, so maybe this was only needed when my code was broken 😄